### PR TITLE
feat : dark-theme and mobile responsive added

### PR DESCRIPTION
**Pull request by mikesch803**
1. add mobile responsiveness 

note: this PR contains only mobile responsiveness code only

### CODE REVIEW

**Review by VLeads — APPROVED** [review]
Code is clean. Great. You can merge it !!

File: src/index.js [hunk 1]
```diff
@@ -18,18 +18,17 @@ const root = ReactDOM.createRoot(document.getElementById("root"));
...
+          <ArchiveProvider>
+            <NoteProvider>
+              <FilterProvider>
+                <AuthProvider>
+                  <App />
+                </AuthProvider>
+              </FilterProvider>
+            </NoteProvider>
+          </ArchiveProvider>
+        </TrashProvider>
```
Comment (dev-ashishsethi) [inline src/index.js:31]: you could have included all of these providers in one

File: src/pages/home/Home.jsx [hunk 2]
```diff
@@ -21,21 +20,23 @@ export function Home() {
...
+const [filterModal, setFilterModal] = useState(false);
+
+
   return (
     <div className="page-layout">
       <Aside />
       <main className="main">
-        <SearchBar/>
+        <div className="row-container"><SearchBar/>
+        <button className="btn btn-primary btn-filter" onClick={()=>setFilterModal(!filterModal)}>filter</button></div>
```
Comment (dev-ashishsethi) [inline src/pages/home/Home.jsx:37]: Use === false instead of !

File: src/pages/signup/Signup.css [hunk 3]
```diff
@@ -1,53 +1,55 @@
...
+  background-color: var(--COLOR-FT);
+  gap: 0.5rem;
+  height: fit-content;
+  border: 1px solid;
+  border-radius: var(--BORDER-RADIUS);
+}
+
+.signup-checkbox {
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
```
Comment (dev-ashishsethi) [inline src/pages/signup/Signup.css:22]: don't use rem for less than 1 rem

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

**Pull request merged**